pkg/services/pluginsintegration/pluginexternal/check.go

```diff
@@ -0,0 +1,43 @@
+package pluginexternal
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/plugins/log"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Service struct {
+	cfg         *setting.Cfg
+	logger      log.Logger
+	pluginStore pluginstore.Store
+}
+
+func ProvideService(
+	cfg *setting.Cfg, pluginStore pluginstore.Store,
+) (*Service, error) {
+	dslogger := log.New("datasources")
+	s := &Service{
+		cfg:         cfg,
+		logger:      dslogger,
+		pluginStore: pluginStore,
+	}
+	return s, nil
+}
+
+func (s *Service) Run(ctx context.Context) error {
+	s.validateExternal()
+	return nil
```

Comment:
Nit. Does it work as expected to return nil directly or do we need `return ctx.Err()` as can be seen in many other background services?